rotkehlchen/rotkehlchen.py

```diff
@@ -170,6 +170,24 @@ def __init__(self, args: argparse.Namespace) -> None:
         self.shutdown_event = gevent.event.Event()
         self.migration_manager = DataMigrationManager(self)
 
+    def kill_running_tx_query_tasks(self) -> None:
+        """Checks for running greenlets related to transactions query and kills them."""
+        assert self.task_manager is not None
+        log.debug(
+            f'Attempting to kill running transaction queries tasks '
+            f'from api task greenlets {self.api_task_greenlets} and '
+            f'task manager running greenlets {self.task_manager.running_greenlets}',
+        )
+        for greenlet in self.api_task_greenlets:
+            if greenlet.args[0].__func__.__name__ == '_get_evm_transactions':
+                greenlet.kill()
+        self.api_task_greenlets = [greenlet for greenlet in self.api_task_greenlets if not greenlet.dead]  # noqa: E501
+
+        for fn, greenlets in list(self.task_manager.running_greenlets.items()):
+            if fn.__name__ == '_maybe_query_evm_transactions':
```

Comment:
no no no. This will kill them all. Specifically check for `evm_manager.transactions.single_address_query_transactions` and the argument being that address. And the different evm manager depending on the chain id

Reply:
`chain id` cannot be used here since `transactions.single_address_query_transactions` calls the parent class directly.

So the `__qualname__` evaluates to `EvmTransactions.single_address_query_transactions` not the chain specific child class